### DISCUSSION TITLE
AttributeError: module 'mxnet' has no attribute 'mod'

### DISCUSSION THREAD

**Discussion by fresh382227905** [apache/incubator-mxnet#20133] [Q&A]
3070显卡，cuda 11.0 mxnet 2.0.0    
出现AttributeError: module 'mxnet' has no attribute 'mod'
请问怎么解决呢

**Comment by Zha0q1** [comment]
thanks for posting. Would you share about you platform, how you installed MXNet, and the script to reproduce the issue?

**Comment by fresh382227905** [comment]
Thanks for your reply.Here are my versions:
Python 3.6&nbsp;
ubuntu 18.04
cuda 11.0
pip 9.0.1
I downloaded the package from&nbsp;https://dist.mxnet.io/python and I chose this version, i.e mxnet_cu110-2.0.0b20210123-py3-none-manylinux2014_x86_64.whl. I used this command: pip3 install mxnet_cu110-2.0.0b20210123-py3-none-manylinux2014_x86_64.whl, It didn't work. Then I Entered my shell, type python3 ;import pip&nbsp;&nbsp;

and&nbsp; I changed the name as&nbsp;
mxnet_cu110-2.0.0b20210123-py3-none-any.whl
It worked, and the version of mxnet&nbsp; is 2.0.0



However, when I run the codes, An error occured, i.e has no attribute mod.module
Looking forward to your reply.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Zhaoqi ***@***.***&gt;; 
发送时间: 2021年4月7日(星期三) 下午2:37
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [apache/incubator-mxnet] AttributeError: module &#39;mxnet&#39; has no attribute &#39;mod&#39; (#20133)





 
thanks for posting. Would you share about you platform, how you installed MXNet, and the script to reproduce the issue?

**Comment by fresh382227905** [comment]
By the way, my graphics card is&nbsp;&nbsp;GeForce RTX 3070. driver's version is 460




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Zhaoqi ***@***.***&gt;; 
发送时间: 2021年4月7日(星期三) 下午2:37
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [apache/incubator-mxnet] AttributeError: module &#39;mxnet&#39; has no attribute &#39;mod&#39; (#20133)





 
thanks for posting. Would you share about you platform, how you installed MXNet, and the script to reproduce the issue?

**Comment by Zha0q1** [comment]
So you were able to pip install the whl file? Then the error message showed up when you import mxnet?

**Comment by fresh382227905** [comment]
I changed the name of my package's name. It can be installed successfuly. when I import mxnet, it works well.However&nbsp; when I use&nbsp;model = mx.mod.Module(symbol=sym, context=ctx, label_names = None), an&nbsp; error occured.
AttributeError: module 'mxnet' has no attribute 'mod'
My mxnet version is 2.0.0




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Zhaoqi ***@***.***&gt;; 
发送时间: 2021年4月7日(星期三) 下午2:58
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [apache/incubator-mxnet] AttributeError: module &#39;mxnet&#39; has no attribute &#39;mod&#39; (#20133)





 
So you were able to pip install the whl file? Then the error message showed up when you import mxnet?

**Reply by Zha0q1** [reply]
I think this is because 2.0 has deprecated the Module api. You might try 1.x releases instead. 1.8 wheels should be available through pip install https://pypi.org/project/mxnet/

**Comment by fresh382227905** [comment]
Thanks.It does work.&nbsp;I should have upgraded my pip version. then I can install mxnet directly by using command ''&nbsp;pip3 install mxnet-cu110==1.8.0 -i HTTPS://mirrors.aliyun.com/pypi/simple/&nbsp;''&nbsp;
that error has been&nbsp; solved,&nbsp;however another error occured,MXNetError: Check failed: compileResult == NVRTC_SUCCESS (5 vs. 0) : NVRTC Compilation failed. Please set environment variable MXNET_USE_FUSION to 0.
nvrtc: error: invalid value for --gpu-architecture (-arch)
I'm going to solve it.
Thanks






------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Zhaoqi ***@***.***&gt;; 
发送时间: 2021年4月7日(星期三) 下午3:29
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [apache/incubator-mxnet] AttributeError: module &#39;mxnet&#39; has no attribute &#39;mod&#39; (#20133)





 
I think this is because 2.0 has deprecated the Module api. You might try 1.x releases instead. 1.8 wheels should be available through pip install https://pypi.org/project/mxnet/
